File: sieve.py
#!/usr/bin/python
import array
import time

if __name__ == "__main__":
	print "Enter the upper limit of prime numbers to calculate: "
	userInput = raw_input()
	n = int(userInput.strip())
	# print "the user entered " + userInput

	startTime = time.time()
	sieveList = []
	for num in range(2,n+2):
		sieveList.append(num)

	i = 0
	while(i < n - 1):
		if((sieveList[i] % (2*(i+1))) == 0):
			k = 0
			while(k < n - 1):
				
				if(k <= n-1 and sieveList[k-2] is not 2):
					sieveList[k-2] = 0
				k = i * 2
				i += 1
		i += 1


	i = 0
	while(i < n - 1):
		if((sieveList[i] % (3*(i + 1))) == 0):
			k = 0
			while(k < n-1):
				if(k <= n-1 and sieveList[k-2 is not 3]):
					sieveList[k-2] = 0
				k = i * 3
				i += 1
		i += 1

	i = 0
	while(i < n - 1):
		if((sieveList[i] % (5*(i + 1))) == 0):
			k = 0
			while(k < n-1):
				if(k < n-1 and sieveList[k-2 is not 5]):
					sieveList[k-2] = 0
				k = i * 5
				i += 1
		i += 1

	fileOutput = open('primeNumberPythonOutput.txt', 'w')
	fileOutput.write("results:\n")
	for num in range(0, n - 3):
		if (sieveList[num] != 0):
			fileOutput.write("%s\n" % sieveList[num])


	fileOutput.close()
	print ("execution time: %s seconds" % (time.time() - startTime))
			

File: sieve.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>


int main()
{

    int upperlimitInt = 0;
    printf("Enter the upper limit of prime numbers to calculate: ");
    scanf("%d", &upperlimitInt);
    int sieveArray[upperlimitInt];
    int newSieveCount = 0;
    int secondCount = 0;
    FILE * fp;

    clock_t initTime = clock();

    for(int i = 2; i <= upperlimitInt; i++)
    {
        sieveArray[i] = i; 
    }

    for(int i = 2; i <= upperlimitInt; i++)
    {
        if(sieveArray[i] % 2*i == 0)
        {
            // printf("shit equals 0? %d\n", sieveArray[i]);

            for(int k = i * 2; k <= upperlimitInt; k += i)
            {
                sieveArray[k] = 0;
            }

            for(int k = i * 3; k <= upperlimitInt; k += i)
            {
                sieveArray[k] = 0;
            }
        }
    }

    for(int i = 2; i <= upperlimitInt; i++)
    {
        if(sieveArray[i] % 3*i == 0 )
        {

            for(int k = i * 3; k <= upperlimitInt; k += i)
            {
                sieveArray[k] = 0;
            }
        }
    }

    for(int i = 2; i <= upperlimitInt; i++)
    {
        if(sieveArray[i] % 5*i == 0)
        {
            for(int k = i * 5; k <= upperlimitInt; k += i)
            {
                sieveArray[k] = 0;
            }
        }
    }

    for (int i = 2; i <= upperlimitInt; i++)
    {
        if(sieveArray[i] != 0)
        {
            newSieveCount++;
        }
    }
    int resultsArray[newSieveCount];

    for(int i = 2; i <= upperlimitInt; i++)
    {
        if(sieveArray[i] != 0)
        {
            resultsArray[secondCount] = sieveArray[i];
            secondCount++;
        }
    }

    fp = fopen("primeNumberCOutput.txt", "w");
    fprintf(fp, "results\n");

    for(int i = 0; i < newSieveCount; i++)
    {
        fprintf(fp, "%d\n", resultsArray[i]);
    }
    
    fprintf(fp, "\n");
    fclose(fp);

    clock_t finishTime = clock();

    printf("Execution Time: %f seconds\n", (float)(finishTime - initTime)/CLOCKS_PER_SEC);

    return 0;
}
